### PR TITLE
[JUJU-4044] Refactor db-accessor worker tests

### PR DESCRIPTION
**Pull request by manadart**
Changes being worked on in another patch proved difficult to test due to poor test suite arrangement in the _worker/dbaccessor_ package.

The following changes improve the elegance of test infrastructure there:
- Tests for name-spaced DB handling are moved to their own suite.
- The `workerSuite` no longer embeds `dbBaseSuite` because it does not need a controller schema or transaction runner.
- Extraneous behaviour triggers are removed from `ensureStartup`. These were copied needlessly from another test.
- `expectNodeStartupAndShutdown` no longer returns a sync channel. `ensureStartup` is used everywhere.
- `expectNodeStartupAndShutdown` does not do conditional expectations of `Handover`. Those are just added where they are needed.
- Various mocks are relocated from base suites to child suites and vice versa depending on where they are required.
- The deferred calls the `CleanKill` are replaced with `Kill` where appropriate. We don't need to call `CleanKill` twice.

## QA steps

`TEST_PACKAGES="./worker/dbaccessor -count=1 -race" make run-go-tests`

### CODE REVIEW

File: worker/dbaccessor/tracker_test.go [hunk 1]
```diff
@@ -77,8 +74,7 @@ func (s *trackedDBWorkerSuite) TestWorkerDBIsNotNil(c *gc.C) {
 
 	w, err := s.newTrackedDBWorker(defaultPingDBFunc)
 	c.Assert(err, jc.ErrorIsNil)
-
-	defer workertest.CleanKill(c, w)
+	defer w.Kill()
```
Comment (SimonRichardson) [inline worker/dbaccessor/tracker_test.go:77]: We should use dirty kill here, so the tests will trace the failure, even though we don't care. Useful for debugging.

```suggestion
	defer workertest.DirtyKill(c, w)
```

File: worker/dbaccessor/tracker_test.go [hunk 2]
```diff
@@ -369,7 +358,7 @@ func (s *trackedDBWorkerSuite) TestWorkerCancelsTxn(c *gc.C) {
 		case <-time.After(testing.ShortWait):
 			c.Fatal("timed out waiting for sync")
 		}
-		workertest.CheckKill(c, w)
+		_ = workertest.CheckKill(c, w)
```
Comment (SimonRichardson) [inline worker/dbaccessor/tracker_test.go:361]: This should be DirtyKill if we don't care about the error, if we do, we should use `CleanKill`.

```suggestion
		workertest.DirtyKill(c, w)
```

File: worker/dbaccessor/tracker_test.go (hunk 1 above)
Comment (manadart) [reply worker/dbaccessor/tracker_test.go:77]: There's no real point. We're doing a `CleanKill` at the end. This is only the first killer of the worker if we fail before, the cause of which will be pertinent output.

Comment (SimonRichardson) [reply worker/dbaccessor/tracker_test.go:77]: > There's no real point.

Consistency is a strong counterargument to this.

Comment (manadart) [reply worker/dbaccessor/tracker_test.go:77]: Indeed. I'd rather use the `workertest` methods anywhere we really need to wait for the worker. In all the places we don't (where it's basically garbage collection) just `Kill`. It means faster test feedback, like when you get "worker not stopping" for a missed mock expectation.

Comment (manadart) [reply worker/dbaccessor/tracker_test.go:77]: Righto. I've done it.

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**